frontend/src/metabase/query_builder/components/filters/modals/BulkFilterItem/constants.ts

```diff
@@ -1,4 +1,5 @@
 export const FIELD_PRIORITY = [
+  "type/Date",
```

Comment:
The problem with `isDate()` is that it returns `true` for all `type/Temporal` types, which includes `type/Time`, and this component doesn't work for just times 😢 

I agree the list of types is a little verbose, but I think it's strength is that it's really simple and easy to follow.

https://github.com/metabase/metabase/blob/ef0d02d307cd65debd5f58ec3cdabd3a2d0bf8f3/shared/src/metabase/types.cljc#L172-L186

To double-check I wrote a test that, unfortunately, passes.

```js
    it('incorrectly flags a time as a date', () => {
      expect(isDate({ base_type: TYPE.Time })).toEqual(true);
      expect(isDate({ base_type: TYPE.Date })).toEqual(true);
      expect(isDate({ base_type: TYPE.DateTime })).toEqual(true);
    });
```

Reply:
I agree this isn't ideal. Actually, I was looking at `isa` initially but found `isXXX` exists, I think like for other fields you might be able to use those e.g. `isNumber`.

I think you might just hit another **one-off** thing with built-in functions 😆 maybe that's why this function exists?

https://github.com/metabase/metabase/blob/be61433b7e8f5aa4236d5bf8cf12355f61fcebbf/frontend/src/metabase/lib/schema_metadata.js#L183-L192

To be clear, I'm fine with you having `FIELD_PRIORITY` at first but as I see you keep adding more and more types to this array I felt like it's getting out of hand, and kind of feel like we're reinventing the wheel. Because what we essentially want to do is check if *a type* corresponds to another *primitive type* and show the filter accordingly.

And to me, that could be achieved with a handful of checks of simpler functions like `isNumber`, or `isa(type, anotherType)`, or other functions that use `isa` under the hood (like `isDateWithoutTime`).

I'll leave this to your discretion then. As I don't know if this `FIELD_PRIORITY` list is going to keep expanding forever.  e.g. If a field with `base_type` of **type/Text** has a `semantic_type` of **type/CreationTimestamp** needs to show a date filter then you probably need to add **type/CreationTimestamp** to the list, and there are like tons of semantic types if we need to support them.

Reply:
You're right about this. thanks for pushing back. I'm working on a better solution. 👍